Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -14482,118 +14482,102 @@ void SpeculativeJIT::compileEnumeratorNextUpdateIndexAndMode(Node* node)
-        JSValueRegsTemporary result(this);
+        GPRTemporary newIndex(this);
+        GPRTemporary newMode(this, Reuse, mode);
         GPRReg baseGPR = base.gpr();
-        JSValueRegs resultRegs = result.regs();
 
         // Has the same structure as the enumerator.
-        load32(Address(baseGPR, JSCell::structureIDOffset()), resultRegs.payloadGPR());
-        speculationCheck(BadCache, JSValueSource(), node, branch32(NotEqual, resultRegs.payloadGPR(), Address(enumeratorGPR, JSPropertyNameEnumerator::cachedStructureIDOffset())));
+        load32(Address(baseGPR, JSCell::structureIDOffset()), newIndex.gpr());
+        speculationCheck(BadCache, JSValueSource(), node, branch32(NotEqual, newIndex.gpr(), Address(enumeratorGPR, JSPropertyNameEnumerator::cachedStructureIDOffset())));
 
-        load32(Address(enumeratorGPR, JSPropertyNameEnumerator::flagsOffset()), resultRegs.payloadGPR());
-        and32(TrustedImm32(JSPropertyNameEnumerator::enumerationModeMask), resultRegs.payloadGPR());
-        speculationCheck(BadCache, JSValueSource(), node, branch32(NotEqual, TrustedImm32(JSPropertyNameEnumerator::OwnStructureMode), resultRegs.payloadGPR()));
+        load32(Address(enumeratorGPR, JSPropertyNameEnumerator::flagsOffset()), newIndex.gpr());
+        and32(TrustedImm32(JSPropertyNameEnumerator::enumerationModeMask), newIndex.gpr());
+        speculationCheck(BadCache, JSValueSource(), node, branch32(NotEqual, TrustedImm32(JSPropertyNameEnumerator::OwnStructureMode), newIndex.gpr()));
 
-        move(indexGPR, resultRegs.payloadGPR());
+        move(indexGPR, newIndex.gpr());
         Jump initMode = branchTest32(Zero, modeGPR);
 
-        add32(TrustedImm32(1), resultRegs.payloadGPR());
+        add32(TrustedImm32(1), newIndex.gpr());
 
         initMode.link(this);
-#if USE(JSVALUE64)
-        or64(TrustedImm64(JSValue::DoubleEncodeOffset | static_cast<uint64_t>(JSPropertyNameEnumerator::OwnStructureMode) << 32), resultRegs.payloadGPR());
-#else
-        move(TrustedImm32(JSPropertyNameEnumerator::OwnStructureMode), resultRegs.tagGPR());
-#endif
 
-        jsValueResult(resultRegs, node);
+        if (m_graph.m_tupleData.at(node->tupleOffset() + 1).refCount)
+            move(TrustedImm32(static_cast<unsigned>(JSPropertyNameEnumerator::OwnStructureMode)), newMode.gpr());
+
+        useChildren(node);
+        strictInt32TupleResultWithoutUsingChildren(newIndex.gpr(), node, 0);
+        if (m_graph.m_tupleData.at(node->tupleOffset() + 1).refCount)
+            strictInt32TupleResultWithoutUsingChildren(newMode.gpr(), node, 1);
         return;
     }
 
     JSValueOperand base(this, baseEdge);
+#if USE(JSVALUE64)
+    GPRTemporary newMode(this, Reuse, mode);
+#endif
     JSValueRegs baseRegs = base.regs();
 
+
     flushRegisters();
     JSValueRegsFlushedCallResult result(this);
     JSValueRegs resultRegs = result.regs();
+    // FIXME: We should teach callOperation how to handle more than one result.
```

Comment:
extractResult doesn't quite work because it does a single register on 64-bit. I guess I could do:
```
setupArguments<decltype(operationEnumeratorNextUpdateIndexAndMode)>(LinkableConstant::globalObject(*this, node), baseRegs, indexGPR, modeGPR, enumeratorGPR);
appendCallSetResult(operationEnumeratorNextUpdateIndexAndMode, resultGPR1, resultGPR2);
```